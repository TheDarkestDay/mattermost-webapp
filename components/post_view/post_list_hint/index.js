import { PostListHint } from './post_list_hint';

export { PostListHint };