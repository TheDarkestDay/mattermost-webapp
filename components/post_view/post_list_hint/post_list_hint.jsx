import PropTypes from 'prop-types';
import React from 'react';

import Toast from 'components/toast/toast';

export const PostListHint = ({show, onDismiss, children}) => {
  return (
    <Toast show={show} extraClasses="toast__hint" onDismiss={onDismiss}> 
        {children}
    </Toast>
  );
};

PostListHint.propTypes = {
  /* Controls whether this tooltip should be shown */
  show: PropTypes.bool.isRequired,

  /* Handler function for Dismiss button */
  onDismiss: PropTypes.func,

  children: PropTypes.node,
};