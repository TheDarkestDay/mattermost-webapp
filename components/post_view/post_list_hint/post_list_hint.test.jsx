import React from 'react';
import {shallow} from 'enzyme';
import { PostListHint } from 'components/post_view/post_list_hint';
import { mountWithIntl } from 'tests/helpers/intl-test-helper';

describe('components/PostListHint', () => {
  const defaultProps = {
    show: true,
    onDismiss: jest.fn(),
  };

  it('should match snapshot when tooltip is visible', () => {
    const wrapper = shallow(<PostListHint {...defaultProps}><b>Hint: you can dismiss this message</b></PostListHint>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when tooltip is hidden', () => {
    const wrapper = shallow(<PostListHint {...{...defaultProps, show: false}}><b>Hint: you can dismiss this message</b></PostListHint>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire dismissHandler when dismiss button is clicked', () => {
    const dismissHandler = jest.fn();
    const wrapper = mountWithIntl(<PostListHint {...{...defaultProps, onDismiss: dismissHandler}}><b>Hint: you can dismiss this message</b></PostListHint>);
    const dismissButton = wrapper.find('.toast__dismiss');

    dismissButton.simulate('click');

    expect(dismissHandler).toBeCalled();
  });
});