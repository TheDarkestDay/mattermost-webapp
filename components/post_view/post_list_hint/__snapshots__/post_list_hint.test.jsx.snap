// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostListHint should match snapshot when tooltip is hidden 1`] = `
<Toast
  extraClasses="toast__hint"
  onDismiss={[MockFunction]}
  show={false}
>
  <b>
    Hint: you can dismiss this message
  </b>
</Toast>
`;

exports[`components/PostListHint should match snapshot when tooltip is visible 1`] = `
<Toast
  extraClasses="toast__hint"
  onDismiss={[MockFunction]}
  show={true}
>
  <b>
    Hint: you can dismiss this message
  </b>
</Toast>
`;
